From 93979fb7e086a55d38921d05ba464a11ae1c22ef Mon Sep 17 00:00:00 2001
From: Yifeng Zhao <yifeng.zhao@rock-chips.com>
Date: Thu, 18 Jun 2020 18:12:11 +0800
Subject: [PATCH] soc: rockchip: mtd_vendor_storage: fix issue data structure
 not align to 4KB

The size of vendor_info need align to 4KB and compatible with uboot.

Signed-off-by: Yifeng Zhao <yifeng.zhao@rock-chips.com>
Change-Id: I16ae2cf5c976032e944bfb590d38ec89831f9b2f
---
 drivers/soc/rockchip/mtd_vendor_storage.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/rockchip/mtd_vendor_storage.c b/drivers/soc/rockchip/mtd_vendor_storage.c
index db2ea11b4b35..dd8ea3abe097 100644
--- a/drivers/soc/rockchip/mtd_vendor_storage.c
+++ b/drivers/soc/rockchip/mtd_vendor_storage.c
@@ -44,7 +44,7 @@ struct vendor_info {
 	u16	free_offset;
 	u16	free_size;
 	struct	vendor_item item[62];
-	u8	data[MTD_VENDOR_PART_SIZE * 512 - 1024 - 8];
+	u8	data[MTD_VENDOR_PART_SIZE * 512 - 512 - 8];
 	u32	hash;
 	u32	version2;
 };
-- 
2.35.3

